web/src/rows.ts

```diff
@@ -90,18 +94,20 @@ export function id($message_row) {
 
     if ($message_row.length !== 1) {
         blueslip.error("Caller should pass in a single row.");
+        return undefined;
     }
 
     const zid = $message_row.attr("zid");
 
     if (zid === undefined) {
         blueslip.error("Calling code passed rows.id a row with no zid attr.");
+        return undefined;
     }
```

Comment:
This contradicts the “don’t return early” comment above. Previously we would have gotten `Number.parseFloat(undefined)`, which is `NaN`. Probably nobody wants `NaN`, but the comment makes me scared about changing it without investigating.

Interestingly, we are seeing both of these errors in Sentry every few days. We should probably debug that.

Reply:
Hm, so `Number.parseFloat` wants a string, but I can return `NaN`, so I guess I'll do that?

Reply:
Oh I can reproduce on chat.zulip.org: in a very long view like **All messages**, send a message, scroll up a lot by pressing `Home` many times, then press `←`.

Reply:
Can that be a separate issue from this PR?

Reply:
Fixed, using Tim's suggestion [here](https://chat.zulip.org/#narrow/stream/464-kandra-js-errors/topic/Error.3A.20Caller.20should.20pass.20in.20a.20single.20row.2E)